src/components/Login-registration-components/Button.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { type ReactElement } from 'react';
+
+interface ButtonProps {
+  type?: 'button' | 'submit' | 'reset';
+  onClick?: () => void;
+  label: string;
+  className?: string;
+  disabled?: boolean;
+  variant?: string;
+}
+
+const Button: React.FC<ButtonProps> = ({
+  type = 'button',
+  onClick,
+  label,
+  className = '',
+  disabled = false,
+}: ButtonProps): ReactElement => {
```

Comment:
```suggestion
}) => {
```
`React.FC<ButtonProps>` already defines its props type and return type, so it's redundant